tools/chapel-py/src/resolution.cpp

```diff
@@ -137,6 +140,19 @@ static const ID scopeResolveResultsForNode(Context* context, const AstNode* node
   return ID();
 }
 
+const resolution::ResolvedExpression* scopeResolveResultsForNode(Context* context, const AstNode* node) {
+  const AstNode* search = node;
+  while (search) {
+    if (auto rr = scopeResolveViaFunction(context, search, node)) {
+      return rr;
+    } else if (auto rr = scopeResolveViaModule(context, search, node)) {
+      return rr;
+    }
+    search = parsing::parentAst(context, search);
+  }
+  return nullptr;
+}
+
```

Comment:
Can this be used as a helper in `scopeResolveToIdForNode`

Reply:
Can some of the common code be pulled out into a helper?

Reply:
Really not, that's what `scopeResolveViaFunction` and `scopeResolveViaModule` are doing. I don't see what else can be pulled out.

Reply:
Perhaps this?
```
    if (auto rr = scopeResolveViaFunction(context, search, node)) {
      return rr;
    } else if (auto rr = scopeResolveViaModule(context, search, node)) {
      return rr;
    }
```
Although at some point there is diminishing returns from pulling out too many helper methods. I think its ok to leave as-is